### PR TITLE
Replace strcpy with memcpy

### PR DESCRIPTION
**Pull request by dcristoloveanu**
Replace strcpy with memcpy

### CODE REVIEW

File: src/connection.c [hunk 1]
```diff
@@ -978,7 +979,8 @@ CONNECTION_HANDLE connection_create2(XIO_HANDLE xio, const char* hostname, const
 
                     if (result != NULL)
                     {
-                        result->container_id = (char*)malloc(strlen(container_id) + 1);
+                        size_t container_id_length = strlen(container_id);
+                        result->container_id = (char*)malloc(container_id_length + 1);
```
Comment (ewertons) [inline src/connection.c]: Would it be too conservative to use (sizeof(char) * container_id_length) ?

Comment (dcristoloveanu) [inline src/connection.c]: Well, it needs to look for the end of the string and there is no need for it in this case, the length is known.

Comment (dcristoloveanu) [inline src/connection.c]: sizeof(char) is always 1. C99 is very crisp about it.
"When applied to an operand that has type char, unsigned char, or signed char,
(or a qualified version thereof) the result is 1."

**Comment by dcristoloveanu** [comment]
Cheers!

**Pull request merged**